packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -169,6 +176,13 @@ class DropdownButton extends React.Component {
       theme.className,
     );
 
+    let buttonAriaLabel = '';
+    const modifiedButtonAttrs = { ...buttonAttrs };
+    if (modifiedButtonAttrs && modifiedButtonAttrs['aria-label']) {
+      buttonAriaLabel = modifiedButtonAttrs['aria-label'];
+      delete modifiedButtonAttrs['aria-label'];
+    }
```

Comment:
does using buttonAttrs like below throws linter error..??
no need to check for (buttonAttrs) since`buttonAttrs` is initialised with defaultValue. (buttonAtrrs) will always evaluate to `true`.

```suggestion
    if (buttonAttrs['aria-label']) {
      buttonAriaLabel = buttonAttrs['aria-label'];
      delete buttonAttrs['aria-label'];
    }
```

Reply:
let buttonAriaLabel = '';
if (buttonAttrs['aria-label']) {
//Now since this condition is true it will enter the if condition
and buttonAttrs['aria-label'] is deleted
But it won't enter this condition the second time which is why buttonAriaLabel will be empty and when the user tabs on the button second time it won't announce the correct aria-label.

That's why I used object spread syntax ({ ...buttonAttrs }) to create a shallow copy of the buttonAttrs object and then modify the copied object. This ensures that the original buttonAttrs object remains unchanged.